File: sample/src/main/java/com/sha/formvalidatorsample/adapter/RecyclerAdapter.java
package com.sha.formvalidatorsample.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.sha.formvalidatorsample.R;

import java.util.List;

public class RecyclerAdapter extends RecyclerView.Adapter<RecyclerAdapter.Vh> {

    private static List<FieldItem> items = FieldInfo.items;

    public class Vh extends RecyclerView.ViewHolder {
       private TextView tvTitle;

        public Vh(@NonNull View itemView) {
            super(itemView);
            tvTitle = itemView.findViewById(R.id.tvTitle);
            itemView.setOnClickListener(__ -> items.get(getAdapterPosition()).showDemo(itemView.getContext()));
        }

        public void onBind(FieldItem item){
            tvTitle.setText(item.title);
        }
    }

    @NonNull
    @Override
    public Vh onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new Vh(LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_example, parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull Vh holder, int position) {
        holder.onBind(items.get(position));
    }

    @Override
    public int getItemCount() {
        return items.size();
    }


}

File: sample/src/main/java/com/sha/formvalidatorsample/classic/adapter/RecyclerAdapter.kt
package com.sha.formvalidatorsample.classic.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

import com.sha.formvalidatorsample.R

class RecyclerAdapter : RecyclerView.Adapter<RecyclerAdapter.Vh>() {

    inner class Vh(itemView: View) : RecyclerView.ViewHolder(itemView) {
        private val tvTitle: TextView = itemView.findViewById(R.id.tvTitle)

        init {
            itemView.setOnClickListener { items[adapterPosition].showDemo(itemView.context) }
        }

        fun onBind(item: FieldItem) {
            tvTitle.text = item.title
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): Vh {
        return Vh(LayoutInflater.from(parent.context)
                .inflate(R.layout.item_example, parent, false))
    }

    override fun onBindViewHolder(holder: Vh, position: Int) = holder.onBind(items[position])

    override fun getItemCount() = items.size

    companion object {
        private val items = FieldInfo.items
    }

}
